sssd_test_framework/roles/base.py

```diff
@@ -189,6 +190,11 @@ def __init__(self, *args, **kwargs) -> None:
         Authentication helpers.
         """
 
+        self.hostnameutils: HostnameUtils = HostnameUtils(self.host)
```

Comment:
maybe add `.postpone_setup()` to avoid running __enter__ code if the utility is not used